src/network/networkd-wiphy.c

```diff
@@ -104,7 +242,89 @@ static int wiphy_update_name(Wiphy *w, sd_netlink_message *message) {
         if (r < 0)
                 return r;
 
-        return hashmap_ensure_put(&w->manager->wiphy_by_name, &string_hash_ops, w->name, w);
+        r = hashmap_ensure_put(&w->manager->wiphy_by_name, &string_hash_ops, w->name, w);
+        if (r < 0)
+                return r;
+
+        return 1;
+}
+
+static int wiphy_update_device(Wiphy *w) {
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        int r;
+
+        assert(w);
+
+        if (!udev_available())
+                return 0;
+
+        w->dev = sd_device_unref(w->dev);
+
+        if (!w->name)
```

Comment:
I wonder if we should simply not allocate a new Wiphy object if it doesn't have a name? I'd move `r = sd_netlink_message_read_string(message, NL80211_ATTR_WIPHY_NAME, &name);` out of `wiphy_update_name()` and into  `manager_genl_process_nl80211_wiphy()` and then if we can't read the wiphy name, ignore the message.

From nl80211.h:

```
 * @NL80211_CMD_NEW_WIPHY: Newly created wiphy, response to get request
 *	or rename notification. Has attributes %NL80211_ATTR_WIPHY and
 *	%NL80211_ATTR_WIPHY_NAME.
```

So I think we can assume that each new wiphy genl message will have a name attr included.

With this change, we can assume `name` is set everywhere else and remove all these `if` statements that check if `w->name` is set.

This can be done in a later PR though.